### PR TITLE
Remove formats property from private posts in content api

### LINKED ISSUE #10118: Remove the formats property from posts in content api
Unless the request is made with a member context, we should remove the formats of any post deemed private

### PR DESCRIPTION
**Pull request by rshbhgrg**
closes #10118 

- Uses post serializer, members lab setting and `jwt` members token to identify and serialize content(`formats`) data, sets it empty in response of content API posts/pages requests. 
- Currently uses hardcoded `#members` tag as identifier for members-only content

### CODE REVIEW

File: core/server/api/v2/utils/serializers/output/utils/members.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const labsUtil = require('../../../../../../services/labs');
+
+const forPost = (attrs, frame) => {
+    const hideMemberOnlyContent = labsUtil.isSet('members') && !frame.context.member;
```
Comment (allouis) [inline core/server/api/v2/utils/serializers/output/utils/members.js]: This should be `frame.original.context.member` :+1:

Comment (m1guelpf) [inline core/server/api/v2/utils/serializers/output/utils/members.js]:
```suggestion
    const hideMemberOnlyContent = labsUtil.isSet('members') && !frame.original.context.member;
```

File: core/server/api/v2/utils/serializers/output/utils/members.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    if (hideMemberOnlyContent) {
+        ['plaintext', 'html'].forEach((field) => {
+            attrs[field] = '';
```
Comment (m1guelpf) [inline core/server/api/v2/utils/serializers/output/utils/members.js]: shouldn't we show some kind of warning or just completely remove the post instead?

Comment (allouis) [reply core/server/api/v2/utils/serializers/output/utils/members.js]: The warning will be handled in the templates for now we think.
e.g.
```hbs
{{#if hasContent}}
  {{content}}
{{else}}
  become a member yo'
{{/if}}
```

**Comment by rishabhgrg** [comment]
@m1guelpf: This PR is still [WIP] and changes are being pushed to it actively, please hold off comments till its out of WIP status

File: core/server/api/v2/utils/serializers/input/posts.js [hunk 3]
```diff
@@ -39,6 +47,7 @@ module.exports = {
             }
             // CASE: the content api endpoint for posts should not return mobiledoc
             removeMobiledocFormat(frame);
+            includeTags(frame);
```
Comment (rishabhgrg) [inline core/server/api/v2/utils/serializers/input/posts.js]: I think we can put this behind `labs.isSet('members')` check, so the additional overhead of including tags in all posts request is minimized to when the members flag is enabled.

File: core/server/api/v2/utils/serializers/output/utils/mapper.js [hunk 4]
```diff
@@ -11,6 +11,9 @@ const mapPost = (model, frame) => {
     if (utils.isContentAPI(frame)) {
         date.forPost(jsonModel);
         members.forPost(jsonModel, frame);
+        if (!frame.original.query.include.includes('tags')) {
+            delete jsonModel.tags;
+        }
```
Comment (rishabhgrg) [inline core/server/api/v2/utils/serializers/output/utils/mapper.js]: Maybe better to move this condition inside the members module used here and behind the flag as above, so its all in once place. 😄

File: core/server/api/v2/utils/serializers/input/posts.js [hunk 5]
```diff
@@ -2,6 +2,7 @@ const _ = require('lodash');
 const debug = require('ghost-ignition').debug('api:v2:utils:serializers:input:posts');
 const url = require('./utils/url');
 const utils = require('../../index');
+const labs = require('../../../../../../services/labs');
```
Comment (allouis) [inline core/server/api/v2/utils/serializers/input/posts.js]:
```suggestion
const labs = require('../../../../../services/labs');
```

**Pull request merged**